src/hotspot/cpu/riscv/riscv.ad

```diff
@@ -5172,6 +5182,28 @@ instruct storeNKlass(iRegN src, memory mem)
   ins_pipe(istore_reg_mem);
 %}
 
+// ============================================================================
+// Prefetch instructions
+// Must be safe to execute with invalid address (cannot fault).
+
+instruct prefetchalloc( memory mem ) %{
+  match(PrefetchAllocation mem);
```

Comment:
PS: Should we also put this under control of option UseZicbop like you do in Prefetch::read/write? Did you checked whether those (prefetchalloc and Prefetch::read/write) will ever be used when AllocatePrefetchStyle is 0 (which is the case when UseZicbop is false).

Reply:
The generation of PrefetchAllocationNode is controlled by AllocatePrefetchStyle. It should be fine if UseZicbop is associated with this option.

Reply:
Added `predicate(UseZicbop)` just in case.